pkg/detectors/azurestorage/azurestorage.go

```diff
@@ -98,6 +111,19 @@ func verifyAzureStorageKey(ctx context.Context, client *http.Client, accountName
 	}
 	defer res.Body.Close()
 
+	// parse container names and append in extra data
+	body, err := io.ReadAll(res.Body)
+	if err == nil {
+		response := storageResponse{}
+		if err := xml.Unmarshal(body, &response); err == nil {
```

Comment:
I would personally invert the conditions so `if err != nil` returns the error, rather than discarding it.

Reply:
The decoding failing is notable because it's reasonable to expect that a `200 OK` response means the key is valid. Personally, I might return the error AND set `verified = true`, but per @rosecodym's [comment](https://github.com/trufflesecurity/trufflehog/pull/2807#discussion_r1594184573) you should return the error and mark verification as failed.

Reply:
@rgmz I believe @rosecodym shared 2 choices. the second one seems to be like ignoring the additional metadata if error occures in parsing and reading the stream as verification is based on Status.OK. Please correct me if my understanding is wrong @rosecodym.

Reply:
Yes, that's correct. There are clearly some general questions to resolve:
* If we can't decode the response body, is that a fatal error?
* How should we handle non-fatal errors?

We need to decide on general answers to these general questions, but until we do, I think it is acceptable to answer them with "no" and "ignore them" respectively, because that represents a minimal change from the current behavior.

(Another option we haven't discussed yet is to redact the errors and put them themselves into `ExtraData`.)